ui-src/NVRApplication.js

```diff
@@ -0,0 +1,290 @@
+  );
+  for (let cameraView of cameraViews) {
+    let url = api.recordingsUrl(
+      cameraView.camera.uuid,
+      selectedRange.startTime90k,
+      selectedRange.endTime90k,
+      videoLength
+    );
+    if (url === cameraView.recordingsUrl) {
+      /*
+       * @TODO: Can this actually happen?
+       */
+      continue; // already in progress, nothing to do.
+    }
+    if (cameraView.recordingsReq !== null) {
+      /*
+       * @TODO: Aborting here does not see right.
+       * If there is another request, it would be because settings changed
+       * and so an abort would leave the UI in a possible inconcistent state.
+       */
+      cameraView.recordingsReq.abort();
+    }
+    cameraView.delayedShowLoading(500);
+    let r = api.request(url);
+    cameraView.recordingsUrl = url;
+    cameraView.recordingsReq = r;
+    r.always(function() {
+      cameraView.recordingsReq = null;
+    });
+    r
+      .then(function(data, status, req) {
+        // Sort recordings in descending order.
+        data.recordings.sort(function(a, b) {
+          return b.startId - a.startId;
+        });
+        console.log('Fetched results > updating recordings');
+        cameraView.recordingsJSON = data.recordings;
+      })
+      .catch(function(data, status, err) {
+        console.log(url, ' load failed: ', status, ': ', err);
+      });
+  }
+}
+
+/**
+ * Setup the calendar for use.
+ *
+ * A CalendarView is established as necessary and then it is initialized
+ * from the camera views. This allows the view to determine what days/dates
+ * are involved and configure datepickers etc.
+ *
+ * This should only change when a new cameras load happens and this function
+ * should be called again.
+ *
+ * We also setup a handler for when the view indicates the user selectable
+ * date range had changed. This is used to fetch new detailed data.
+ *
+ * @param  {Iterable} views     Camera views
+ * @param  {Number} videoLength Desired length of video segments
+ * @return {CalendarView}       The camera view to be used
+ */
+function setupCalendar(views) {
+  calendarView.initializeWith(views);
+}
+
+/**
+ * Initialize the page after receiving camera data.
+ *
+ * Sets the following globals:
+ * zone - timezone from data received
+ * cameraViews - array of views, one per camera
+ *
+ * Builds the dom for the left side controllers
+ *
+ * @param  {[type]} data [description]
+ * @return {[type]}      [description]
+ */
+function onReceivedCameras(data) {
+  newTimeZone(data.timeZoneName);
+
+  // Set up controls and values
+  const nvrSettingsView = new NVRSettingsView();
+  nvrSettingsView.onVideoLengthChange = (vl) =>
+    fetch(calendarView.selectedRange, vl);
+  nvrSettingsView.onTimeFormatChange = (format) => {
+    cameraViews.forEach((view) => (view.timeFormat = format));
+  };
+  nvrSettingsView.onTrimChange = (t) => {
+    console.log('Trim handler');
+    const newTrim = e.target.checked;
```

Comment:
```
Uncaught ReferenceError: e is not defined
    at NVRSettingsView.nvrSettingsView.onTrimChange [as _trimHandler] (nvr.bundle.js:31105)
    at HTMLInputElement.trimEl.change (nvr.bundle.js:33506)
    at HTMLInputElement.dispatch (nvr.bundle.js:17011)
    at HTMLInputElement.elemData.handle (nvr.bundle.js:16819)
```

Reply:
Hmm. My browser did not seem to produce this message. Maybe my configuration prevents it. Either way, it is a coding error, and has been fixed.